builtin/credential/userpass/path_user_password.go

```diff
@@ -59,14 +65,34 @@ func (b *backend) pathUserPasswordUpdate(
 
 func (b *backend) updateUserPassword(req *logical.Request, d *framework.FieldData, userEntry *UserEntry) (error, error) {
 	password := d.Get("password").(string)
-	if password == "" {
+	prehashedPassword := d.Get("password_hash").(string)
+
+	if password != "" && prehashedPassword != "" {
+		return fmt.Errorf("can't provide both password and password_hash"), nil
+	}
+
+	if password == "" && prehashedPassword == "" {
 		return fmt.Errorf("missing password"), nil
```

Comment:
This should probably be updated to say that one or the other must be provided, not only mention the password parameter. Overall while the logic in this function looks fine, it might be better organized with a `switch` statement.